test/test_loaders.py

```diff
@@ -38,6 +39,22 @@ def test_base_url_override(openapi3_schema_url, url):
     assert endpoint.base_url == "http://example.com"
 
 
+def test_port_override(openapi3_schema_url):
+    schema = schemathesis.from_uri(openapi3_schema_url, port=8081)
+    endpoint = next(schema.get_all_endpoints())
+    assert endpoint.base_url == "http://127.0.0.1:8081/schema.yaml"
+
+
+def test_port_override_with_ipv6(openapi3_schema_url):
+    url = URL(openapi3_schema_url)
+    parts = list(map(int, url.host.split(".")))
+    ipv6_host = "2002:{:02x}{:02x}:{:02x}{:02x}::".format(*parts)
+    ipv6 = url.with_host("[%s]" % ipv6_host)
+    schema = schemathesis.from_uri(ipv6, port=8081)
```

Comment:
Oh, I missed that the build is failing on this line (other failures come from a flaky test).

I think it could be enough to use mocking here instead of adapting the server. Something like

`mocked = mocker.patch("requests.request", wraps=simple_openapi)`

In the example above, the `simple_openapi` variable is the fixture defined in the root conftest file